LogicLayer/Groups/GroupsService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GameBoard.DataLayer.Entities;
+using GameBoard.DataLayer.Repositories;
+using GameBoard.LogicLayer.Configurations;
+using GameBoard.LogicLayer.Groups.Dtos;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using GameBoard.LogicLayer.Groups.Exceptions;
+
+namespace GameBoard.LogicLayer.Groups
+{
+    public class GroupsService : IGroupsService
+    {
+        private readonly IGameBoardRepository _repository;
+        private GroupsConfiguration GroupsOptions { get; }
+
+        public GroupsService(IGameBoardRepository repository, IOptions<GroupsConfiguration> groupsOptions)
+        {
+            _repository = repository;
+            GroupsOptions = groupsOptions.Value;
+        }
+
+        public async Task AddGroupAsync(string userName, string groupName)
+        {
+            var user = _repository.ApplicationUsers.Single(ApplicationUser.UserNameEquals(userName));
+
+            var group = new Group
+            {
+                Name = groupName,
+                OwnerId = user.Id,
+                Owner = user
+            };
+
+            _repository.Groups.Add(group);
+            await _repository.SaveChangesAsync();
+        }
+
+        public async Task AddUserToGroupAsync(string userName, int groupId)
+        {
+            var user = _repository.ApplicationUsers.Single(ApplicationUser.UserNameEquals(userName));
+            var group = _repository.Groups.Single(g => g.Id == groupId);
+            var groupUserAlreadyIn = _repository.GroupUser.Where(gu => gu.UserId == user.Id && gu.GroupId == group.Id);
+
+            if (group.OwnerId == user.Id)
+            {
+                throw new GroupsException("You cannot add yourself to your group.");
+            }
+            if (groupUserAlreadyIn.Any())
```

Comment:
1. Use `AnyAsync`
2. You can move that to line 44 and call the variable `isUserAlreadyInThisGroup` which is more descriptive.

Reply:
Marking as done.